### PR TITLE
add android implementation.

### PR DESCRIPTION
**Pull request by AndrewJack**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -2,6 +2,11 @@
 #
 # gitignore contributors: remember to update Global/Xcode.gitignore, Objective-C.gitignore & Swift.gitignore
 
+## Android Studio
+.idea/
+android.iml
+local.properties
```
Comment (AndrewJack) [inline .gitignore]: change above to 
```
*.iml
.idea
.gradle
local.properties
```

File: android/build.gradle [hunk 2]
```diff
@@ -0,0 +1,15 @@
+apply plugin: 'com.android.library'
+
+android {
+    compileSdkVersion 23
```
Comment (AndrewJack) [inline android/build.gradle]: 25

File: android/build.gradle [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+    buildToolsVersion "23.0.3"
```
Comment (AndrewJack) [inline android/build.gradle]: 25.0.2

File: android/build.gradle [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+    defaultConfig {
+        minSdkVersion 16
+        targetSdkVersion 22
```
Comment (AndrewJack) [inline android/build.gradle]: 25

File: android/src/main/AndroidManifest.xml [hunk 5]
```diff
@@ -0,0 +1,2 @@
+<manifest xmlns:android="http://schemas.android.com/apk/res/android" package="com.drivetribe.AppShortcuts">
+</manifest>
```
Comment (AndrewJack) [inline android/src/main/AndroidManifest.xml]: delete this

File: android/src/main/java/com/drivetribe/AppShortcuts/AppShortcutsModule.java [hunk 6]
```diff
@@ -0,0 +1,193 @@
...
+import com.facebook.react.bridge.ReactMethod;
+import com.facebook.react.bridge.ReadableArray;
+import com.facebook.react.bridge.ReadableMap;
+import com.facebook.react.bridge.WritableMap;
+import com.facebook.react.modules.core.DeviceEventManagerModule;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class AppShortcutsModule extends ReactContextBaseJavaModule {
```
Comment (AndrewJack) [inline android/src/main/java/com/drivetribe/AppShortcuts/AppShortcutsModule.java]: Add 

```
@ReactModule(name = AppShortcutsModule.REACT_NAME)
```

File: android/src/main/java/com/drivetribe/AppShortcuts/AppShortcutsModule.java [hunk 7]
```diff
@@ -0,0 +1,193 @@
...
+        getShortcutManager().removeAllDynamicShortcuts();
+        mShortcutItems = null;
+    }
+
+    @ReactMethod
+    public void isSupported(com.facebook.react.bridge.Callback callback) {
+        if (callback != null) {
+            callback.invoke(null, Build.VERSION.SDK_INT >= 25);
+        }
+    }
```
Comment (AndrewJack) [inline android/src/main/java/com/drivetribe/AppShortcuts/AppShortcutsModule.java]: delete and implement in JS

File: android/src/main/java/com/drivetribe/AppShortcuts/AppShortcutsModule.java [hunk 8]
```diff
@@ -0,0 +1,193 @@
...
+            @Override
+            public void onNewIntent(Intent intent) {
+                sendJSEvent(intent);
+            }
+        });
+    }
+
+    @Override
+    public String getName() {
+        return "ReactAppShortcuts";
```
Comment (AndrewJack) [inline android/src/main/java/com/drivetribe/AppShortcuts/AppShortcutsModule.java]: Use REACT_NAME

File: android/src/main/java/com/drivetribe/AppShortcuts/AppShortcutsModule.java [hunk 9]
```diff
@@ -0,0 +1,193 @@
...
+        }
+
+        public WritableMap toWritableMap() {
+            WritableMap map = Arguments.createMap();
+            map.putString("url", url);
+            return map;
+        }
+    }
+
+    static class ShortcutItem {
```
Comment (AndrewJack) [inline android/src/main/java/com/drivetribe/AppShortcuts/AppShortcutsModule.java]: extract into separate class

File: index.android.js [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+   */
+  clearShortcutItems: function() {
+    ReactAppShortcuts.clearShortcutItems();
+  },
+
+  /**
+   * Check if quick actions are supported
+   */
+   isSupported: function(callback) {
+     ReactAppShortcuts.isSupported(callback);
```
Comment (AndrewJack) [inline index.android.js]: implement in js

**Comment by huangmr** [comment]
create separate PR for drive tribe

**Pull request closed without merging**